sycl/tools/get_device_count_by_type.cpp

```diff
@@ -61,54 +64,29 @@ int main(int argc, char* argv[]) {
     }
 #endif  // USE_PI_CUDA
 
-    cl_device_type device_type;
+    info::device_type device_type;
     if (type == "cpu") {
-        device_type = CL_DEVICE_TYPE_CPU;
+      device_type = info::device_type::cpu;
     } else if (type == "gpu") {
-        device_type = CL_DEVICE_TYPE_GPU;
+      device_type = info::device_type::gpu;
     } else if (type == "accelerator") {
-        device_type = CL_DEVICE_TYPE_ACCELERATOR;
+      device_type = info::device_type::accelerator;
     } else if (type == "default") {
-        device_type = CL_DEVICE_TYPE_DEFAULT;
+      device_type = info::device_type::automatic;
     } else if (type == "all") {
-        device_type = CL_DEVICE_TYPE_ALL;
+      device_type = info::device_type::all;
     } else  {
         std::cout << "0:Incorrect device type." << std::endl
             << help << std::endl;
         return 0;
     }
 
-    cl_int iRet = CL_SUCCESS;
-    cl_uint platformCount = 0;
-
-    iRet = clGetPlatformIDs(0, nullptr, &platformCount);
-    if (iRet != CL_SUCCESS) {
-        if (iRet == CL_PLATFORM_NOT_FOUND_KHR) {
-            std::cout << "0:OpenCL runtime not found " << std::endl;
-        } else {
-            std::cout << "0:A problem at calling function clGetPlatformIDs count "
-                << iRet << std::endl;
-        }
-        return 0;
-    }
-
-    std::vector<cl_platform_id> platforms(platformCount);
-
-    iRet = clGetPlatformIDs(platformCount, &platforms[0], nullptr);
-    if (iRet != CL_SUCCESS) {
-        std::cout << "0:A problem at when calling function clGetPlatformIDs ids " << iRet << std::endl;
-        return 0;
-    }
-
-    for (cl_uint i = 0; i < platformCount; i++) {
-        cl_uint deviceCountPart = 0;
-        iRet = clGetDeviceIDs(platforms[i], device_type, 0, nullptr, &deviceCountPart);
-        if (iRet == CL_SUCCESS) {
-            deviceCount += deviceCountPart;
-        }
+    std::vector<platform> platforms(platform::get_platforms());
```

Comment:
Don't think it's right approach to use SYCL here.
Instead I suggest removing get_device_count_by_type completely and updating tests so they try to run on all device types so if such a device is not available just print "Skip".

Reply:
Agree on not using SYCL RT. For example, if there is a bug in platforms or device implementation, or something triggered by the plugin discovered, all test will fail and could be very difficult to figure out what the problem is.

Reply:
There is no reason to run all SYCL LIT testing if SYCL RT platforms/device discovery is broken, i.e. can't detect correctly what's there. I think that for the sake of SYCL LIT testing we should run it on all platforms/devices detected by SYCL RT. There should be a separate single test that verifies that SYCL RT is doing the correct discovery as compared to what's discovered by low-level run-times.

Reply:
> To check that the plugin functionality or platform/device implementations are working fine, we can add a test case which checks the listed devices by calling all low level APIs and then checking it w.r.t. the devices returned from plugin. This doesn't need to be checked by all tests.

Seems to me that, somehow, that test should run before any other, so that if that code is broken, no other test is run and we avoid a massive list of failure which can waste a lot of time.

Reply:
I am not sure if LIT has a mode to stop all further testing if a specific test breaks.